apps/emqx_auth_jwt/src/emqx_authn_jwt.erl

```diff
@@ -53,22 +57,18 @@ update(#{use_jwks := false} = Config, _State) ->
     create(Config);
 update(
     #{use_jwks := true} = Config,
-    #{jwk_resource := ResourceId} = State
+    #{jwk_resource := ResourceId}
 ) ->
     case emqx_resource:simple_sync_query(ResourceId, {update, connector_opts(Config)}) of
         ok ->
-            case maps:get(verify_claims, Config, undefined) of
```

Comment:
The problem was here, we selectively updated only resource and `verify_claims`. In the other function cases, we fully recreated the authenticator.